code/modules/hydroponics/compost_bin.dm

```diff
@@ -0,0 +1,167 @@
+#define BASE_BIOMASS_CAPACITY 1500
+#define BASE_COMPOST_CAPACITY 1500
+#define SOIL_COST 25
+#define DECAY 0.2
+#define MIN_CONVERSION 10
+/**
+  * # compost bin
+  * used to make soil from plants.
+  * Doesn't have components.
+  */
+/obj/machinery/compost_bin
+	name = "compost bin"
+	desc = "A wooden bin for composting."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "compost_bin-empty"
+	power_state = NO_POWER_USE
+	density = TRUE
+	anchored = TRUE
+	/// stage in the process, for timing purposes.
+	var/process_counter = 0
+	/// amount of biomass in the compost bin
+	var/biomass = 0
+	/// amount of compost in the compost bin
+	var/compost = 0
+	/// The maximum amount of biomass the compost bin can store.
+	var/biomass_capacity = BASE_BIOMASS_CAPACITY
+	/// The maximum amount of compost the compost bin can store.
+	var/compost_capacity = BASE_COMPOST_CAPACITY
```

Comment:
I do want to revisit Jimkil3's comment here. This will make it weirder for later subtypes, and this is the only thing the define is used for. This is a perfectly acceptable moment to use a "magic number" for the max capacities of biomass and compost!

Reply:
will do.